OpenGLTutorial10/OpenGLTutorial10/Font.cs

```diff
@@ -195,16 +190,20 @@ public FontVAO CreateString(ShaderProgram program, string text, Justification ju
                 // grab the character, replacing with ' ' if the character isn't loaded
                 Character ch = characters[characters.ContainsKey(text[(int)i]) ? text[(int)i] : ' '];
 
-                vertices[i * 4 + 0] = new Vector3(xpos, ch.height, 0);
-                vertices[i * 4 + 1] = new Vector3(xpos, 0, 0);
-                vertices[i * 4 + 2] = new Vector3(xpos + ch.width, ch.height, 0);
-                vertices[i * 4 + 3] = new Vector3(xpos + ch.width, 0, 0);
-                xpos += (int)ch.width;
+                vertices[i * 4 + 0] = new Vector3(xpos + ch.bearing.X,- ch.bearing.Y, 0);
```

Comment:
Changing this line (and the following lines) to add the Height to the y position will better align the resulting vertices with the previous code.

Reply:
Yes, there seems to be some vertical displacement regardless of font, however I do not know which 'Height' you refer to.
If you meant _ch.size.Y_, then this breaks the vertical alignment of the non-monospace font, as characters can have differing vertical sizes and bearings.
_FontTexture.Size.Height_ adjusts the position by too much in my case (about 500 pixels), as I use a large texture.

Reply:
I mean the actual Height property of the BMFont.

`                vertices[i * 4 + 0] = new Vector3(xpos + ch.bearing.X, Height - ch.bearing.Y, 0);
                vertices[i * 4 + 1] = new Vector3(xpos + ch.bearing.X, Height - (ch.bearing.Y + ch.size.Y), 0);
                vertices[i * 4 + 2] = new Vector3(xpos + ch.bearing.X + ch.size.X, Height - ch.bearing.Y, 0);
                vertices[i * 4 + 3] = new Vector3(xpos + ch.bearing.X + ch.size.X, Height - (ch.bearing.Y + ch.size.Y), 0);`

Reply:
My apologies, I did indeed miss it when reviewing your suggestion.
As you asked, I have added the Height to the vertices, and it does align better while still working with non-monospace fonts.